docs/content-schema-requires-media-type.md

```diff
@@ -0,0 +1,28 @@
+---
+title: \`contentSchema\` without \`contentMediaType\` is unhelpful
+code: content-schema-requires-media-type
+categories: readability, style
+dialects: 2019-09, 2020-12, draft7
+autofixable: false
+---
+
+## Description
+`contentSchema` refines the structure of non-JSON media; without a media type consumers cannot choose a codec.
+
+> **Message shown to user:**
+> Add `contentMediaType` or drop `contentSchema`.
+
+### Example 1
+<details><summary>Before</summary>
+```json
+{
+  "type": "string",
+  "contentSchema": {
+    "type": "object"
+  }
```

Comment:
Best practice for `contentSchema` is to include `$id` and `$schema`. It's probably a good idea to model that behavior in this example and also create another rule to check for that as well.

The next version of the spec will say,

> Note that evaluating the `contentSchema` subschema in-place (i.e. as part of its parent schema) will ensure that it is correctly processed. Independent use of the extracted subschema (as returned in an annotation) is only safe if the subschema is an embedded resource which defines both a `$schema` and an absolute IRI `$id`.